src/Microsoft.OData.Edm/Schema/Interfaces/IEdmTargetPath.cs

```diff
@@ -0,0 +1,22 @@
+//---------------------------------------------------------------------
+// <copyright file="IEdmTargetPath.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Microsoft.OData.Edm.Vocabularies;
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// Represents a path to a model element.
+    /// </summary>
+    public interface IEdmTargetPath : IEdmVocabularyAnnotatable

```

Comment:
maybe implment IEdmFullNameElement interface

Reply:
Why?

Reply:
We can use this interface to get the "fullPathString"

Reply:
I opted to cache the Path string.